static/app/views/performance/vitalDetail/vitalsComparison.tsx

```diff
@@ -0,0 +1,227 @@
+import React, {useEffect} from 'react';
+import styled from '@emotion/styled';
+import {Location} from 'history';
+
+import {
+  INDUSTRY_STANDARDS,
+  MIN_VITAL_COUNT_FOR_DISPLAY,
+  SENTRY_CUSTOMERS,
+} from 'sentry/components/performance/vitalsAlert/constants';
+import QuestionTooltip from 'sentry/components/questionTooltip';
+import Tag from 'sentry/components/tag';
+import {t, tct} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Organization} from 'sentry/types';
+import trackAdvancedAnalyticsEvent from 'sentry/utils/analytics/trackAdvancedAnalyticsEvent';
+import EventView from 'sentry/utils/discover/eventView';
+import {WebVital} from 'sentry/utils/discover/fields';
+import VitalsCardDiscoverQuery from 'sentry/utils/performance/vitals/vitalsCardsDiscoverQuery';
+import {webVitalMeh, webVitalPoor} from 'sentry/views/performance/vitalDetail/utils';
+
+type Score = 'poor' | 'meh' | 'good';
+
+type ViewProps = Pick<
+  EventView,
+  'environment' | 'project' | 'start' | 'end' | 'statsPeriod'
+>;
+
+type Props = ViewProps & {
+  location: Location;
+  organization: Organization;
+  vital: WebVital | WebVital[];
+};
+
+const SUPPORTED_VITALS = ['measurements.fcp', 'measurements.lcp'];
+
+function getScore(vital: WebVital, value: number): Score {
+  const poorScore = webVitalPoor[vital];
+  const mehScore = webVitalMeh[vital];
+  if (value > poorScore) {
+    return 'poor';
+  }
+  if (value > mehScore) {
+    return 'meh';
+  }
+  return 'good';
+}
+
+function getIndicatorString(score: Score) {
+  switch (score) {
+    case 'poor':
+      return t('Poor');
+    case 'meh':
+      return t('Meh');
+    default:
+      return t('Good');
+  }
+}
+
+function getTagLevel(score: Score) {
+  switch (score) {
+    case 'poor':
+      return 'error';
+    case 'meh':
+      return 'warning';
+    default:
+      return 'success';
+  }
+}
+
+function MetricsCard({
+  title,
+  vital,
+  value,
+  tooltip,
+}: {
+  title: string;
+  tooltip: string;
+  value: number;
+  vital: WebVital;
+}) {
+  // round to 2 decimals if <10s, otherwise use just 1 decimal
+  const score = getScore(vital, value);
+  const numDecimals = value >= 10_000 ? 1 : 2;
+  const timeInSeconds = value / 1000.0;
+  return (
+    <MetricsCardWrapper>
+      <MetricsTitle>
+        {title} (p75) <StyledQuestionTooltip title={tooltip} size="xs" />
+      </MetricsTitle>
+      <ScoreWrapper>
+        <ScoreContent>{timeInSeconds.toFixed(numDecimals)}s</ScoreContent>
+        <TagWrapper>
+          <StyledTag type={getTagLevel(score)}>{getIndicatorString(score)}</StyledTag>
+        </TagWrapper>
+      </ScoreWrapper>
+    </MetricsCardWrapper>
+  );
+}
+
+function ContentWrapper({
+  organization,
+  vital,
+  children,
+  count,
+}: {
+  children: React.ReactNode;
+  count: number;
+  organization: Organization;
+  vital: WebVital;
+}) {
+  useEffect(() => {
+    trackAdvancedAnalyticsEvent('performance_views.vital_detail.comparsion_viewed', {
```

Comment:
can this hook be moved to `VitalsComparison`? That way we don't need to create additional nesting of components by creating components that don't have visual logic in them

Reply:
I'm aware we use that pattern for API requests and to perform other operations but at least in the perf team we want to minimize that pattern in our components

Reply:
@udameli It's actually difficult to put this component at the higher level for a number of reasons:
1. I'm using data that's provided for by the child function of `VitalsCardDiscoverQuery`
2. I'd have to add extra logic when _not_ to call `trackAdvancedAnalyticsEvent` in `useEffect` because I don't want to call the function if we don't have a vital. As far as I know, I'd have to write code like this:

````
function MyComponent({vital}) {
   useEffect(() => {
    if (!vital) {
        return;
     }
     trackAdvancedAnalyticsEvent('some_event');
   });
    if (!vital) {
        return null;
     }
   // component code
}
```

Much easier to just embed the `useEffect` into the component that is only mounted when we meet all the checks to render the component.

Reply:
Yeah, this seems like more of a consequence of us having to use query components like `VitalCardDiscoverQuery`, which we can address and cleanup when we refactor to using hooks instead

Reply:
hm can a dependency array be useful in a situation like this? but sure if this is easier 👍